### PR TITLE
Add CallBehaviorAction

### PR DESCRIPTION
**Pull request by marek-piirikivi**
### PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I have read, and I am following the [Contributor guide](https://github.com/gaphor/gaphor/blob/main/CONTRIBUTING.md)
- [x] I have read, and I understand the GNOME [Code of Conduct](https://wiki.gnome.org/Foundation/CodeOfConduct)

### PR Type
What kind of change does this PR introduce?

- [ ] Bug fix
- [x] Feature
- [ ] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?

Issue Number: #2464 

### Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by danyeaw** [comment]
This is an awesome contribution, thanks @marek-piirikivi! This could be a follow-up PR, but do you think there should be a way to go to the selected Behavior from the CallBehavior?

**Review by amolenaar — APPROVED** [review]
Excellent!

The [Invocation Actions diagram](https://gaphor--2495.org.readthedocs.build/en/2495/models/uml/10._Actions.html#invocation-actions) also looks really nice.

File: gaphor/UML/actions/action.py [hunk 1]
```diff
@@ -32,6 +32,64 @@ def __init__(self, diagram, id=None):
...
+            draw=self.draw_border_with_fork,
+        )
+
+        self.watch("subject[CallBehaviorAction].behavior.name")
+
+    def name(self):
+        """Name conforms to UML2.5.1 16.3.4.1 naming description"""
+        if self.subject.behavior and not self.subject.name:
+            return self.subject.behavior.name
+        return self.subject.name or ""
```
Comment (amolenaar) [inline gaphor/UML/actions/action.py:61]: One thing to improve it even more is to move this code to a `format` function in `gaphor/UML/umlfmt.py`. Then the activity name will also be used as fallback in the model browser.

Just noticed this feature is not mentioned in the documentation :see_no_evil:

**Comment by marek-piirikivi** [comment]
> Excellent!
> 
> The [Invocation Actions diagram](https://gaphor--2495.org.readthedocs.build/en/2495/models/uml/10._Actions.html#invocation-actions) also looks really nice.

@amolenaar I tried to be as authentic to the UML2.5.1 document and capture everything that seemed relevant. I noticed that Action itself already defines relationship to the InputPin and OutputPin; and so does CallAction+InvocationAction. So, the references are now in uml.py duplicated. :thinking: I am not sure how to address this; or what to do in this instance. Seems as if it is just redefinition of the same thing to use different names for the same thing - in CallAction context, all input pins should be called arguments. What do you think?

**Comment by marek-piirikivi** [comment]
> This is an awesome contribution, thanks @marek-piirikivi! This could be a follow-up PR, but do you think there should be a way to go to the selected Behavior from the CallBehavior?

Yes. I think so. It would be a good follow-up.

**Comment by amolenaar** [comment]
> > Excellent!
> > The [Invocation Actions diagram](https://gaphor--2495.org.readthedocs.build/en/2495/models/uml/10._Actions.html#invocation-actions) also looks really nice.
> 
> @amolenaar I tried to be as authentic to the UML2.5.1 document and capture everything that seemed relevant. I noticed that Action itself already defines relationship to the InputPin and OutputPin; and so does CallAction+InvocationAction. So, the references are now in uml.py duplicated. thinking I am not sure how to address this; or what to do in this instance. Seems as if it is just redefinition of the same thing to use different names for the same thing - in CallAction context, all input pins should be called arguments. What do you think?

I have no clear strategy for those. As long as a relation is not actively used by the application I nowadays tend to leave it out.

The UML meta model is not always consistent. Often a value subsets a value that is not a derived unions; something that our Gaphor model cannot deal with and will ignore.

File: gaphor/UML/actions/action.py (hunk 1 above)
Comment (marek-piirikivi) [reply gaphor/UML/actions/action.py:61]: I just noticed that there is a slight conflict with the name formatting when looking at SysMLv1.6. SysML requires activity name, and optionally you can also include the action name. How should we proceed?

image

Comment (amolenaar) [reply gaphor/UML/actions/action.py]: We can keep is as is for now.

**Review by amolenaar — APPROVED** [review]

**Comment by amolenaar** [comment]
Let's call it a day and merge this. A really well executed PR.

Thanks @marek-piirikivi!

**Pull request merged**